### ISSUE TITLE
False auto-correction in Lint/BlockAlignment

### ISSUE THREAD

**Issue by lumeet** [bbatsov/rubocop#1573]
This kind of code correcly triggers an offense but no auto-correction is applied, even though "[Corrected]" is reported.

```ruby
a = test do
      do_something
    end
```

**Comment by lumeet** [comment]
I was about to fix this but then stopped to wonder what exactly is the correct behavior. The style guide doesn't seem to cover this specific case. On the other hand, the specs imply that this actually should be valid, too. So, either there should be no offense in the first place or auto-correction should actually remove the superfluous whitespace.

**Comment by jonas054** [comment]
The `BlockAlignment` is not clearly described in its comment header or in the specs, and it contains at least one bug, as you have discovered. Here's how it's supposed to work (I think):
* For assignments, `end` should be aligned with the left-hand-side.
* For other expressions, `end` should be aligned with the expression taking the block.
* It's always OK to align `end` with the start of the line where `do` is.

If you want to tackle this, feel free to do so @lumeet. Otherwise I'll do it.

**Comment by bbatsov** [comment]
@lumeet why did you close this?

**Comment by lumeet** [comment]
@bbatsov Isn't it completely solved by #1576? I thought it was so obvious that I didn't bother commenting. I've now learned to use the correct PR header. :)
